language/move-core/types/src/effects.rs

```diff
@@ -0,0 +1,46 @@
+use crate::{
+    account_address::AccountAddress,
+    identifier::Identifier,
+    language_storage::{StructTag, TypeTag},
+};
+use anyhow::Result;
+use std::collections::btree_map::{self, BTreeMap};
+
+/// A collection of changes to modules and resources under a Move account.
+#[derive(Clone)]
+pub struct AccountChangeSet {
+    pub modules: BTreeMap<Identifier, Option<Vec<u8>>>,
+    pub resources: BTreeMap<StructTag, Option<Vec<u8>>>,
+}
+
+impl AccountChangeSet {
+    pub fn squash(&mut self, other: Self) -> Result<()> {
```

Comment:
Given that this + the other `squash` is only used in `vm_genesis`, I wonder if we could implement it there instead? I have some reservations about exposing this as a utility function because the behavior when `self` deletes a module/resource and `other` adds it (or vice versa) is potentially surprising... It would be easy to make the mistake of thinking that this is commutative.

Reply:
Sounds like a good idea in general. I already ended up making this function stricter in a subsequent PR (coming soon) due to similar concerns. Let me try to finalize that one and see what exactly is the best way to handle `squash`.

Reply:
Just realized I don't immediately need `squash` for the incoming PR (#5972). What do you think about just deleting it for now? I can add it back once I start working on the new testing libraries that actually need it and by that time I should understand the desired behavior much better than now.